drivers/wifi/nrf700x/osal/hw_if/hal/src/hal_api.c

```diff
@@ -221,8 +209,17 @@ unsigned long wifi_nrf_hal_buf_map_tx(struct wifi_nrf_hal_dev_ctx *hal_dev_ctx,
 		(desc_id * hal_dev_ctx->hpriv->cfg_params.max_tx_frm_sz) +
 		hal_dev_ctx->hpriv->cfg_params.tx_buf_headroom_sz;
 
+	if (frame_id == 0) {
+		hal_dev_ctx->addr_last_buf = bounce_buf_addr;
+	} else {
+		bounce_buf_addr = hal_dev_ctx->addr_last_buf;
+	}
+
 	rpu_addr = RPU_MEM_PKT_BASE + (bounce_buf_addr - hal_dev_ctx->addr_rpu_pktram_base);
 
+	/* Align the memory */
+	buf_len = (buf_len + 4) & 0xfffffffc;
```

Comment:
if Aligning length enough or do we need to align the buffer address as well?

Reply:
`spim_addr_check` checks both address and length, so, maybe in your tests only length needs alignment, but what about in general or for QSPI? We need to meet the alignment requirements without depending on tests as we cannot excite all scenarios.